gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala

```diff
@@ -364,6 +364,18 @@ case class GremlinScala[End, Labels <: HList](traversal: GraphTraversal[_, End])
     GremlinScala[A, Labels](traversal.union(mappedTraversals: _*))
   }
 
+  def choose[A, B](
+    predicate: GremlinScala[End, HNil] ⇒ GremlinScala[A, _],
+    trueChoice: GremlinScala[End, HNil] ⇒ GremlinScala[B, _],
+    falseChoice: GremlinScala[End, HNil] ⇒ GremlinScala[B, _]): GremlinScala[B, Labels] = {
+    val p: Traversal[_, A] = predicate(start).traversal
```

Comment:
shouldn't this be `predicate: GremlinScala[End, HNil] ⇒ GremlinScala[Boolean, _]` as this is a predicate? Then you only need one type parameter